citi-onboarding/lgbtqia
# Feature/personalidades

## What I did

- create Personality model (with nullable columns)
- create PersonalityController and CRUD. Post and put don't demand nullable columns to be defined. Put replaces the value of an undefined nullable column with null, so as to not function like a patch.
- add respective routes

## How to test

0. If you don't have this repository in your computer:
   1. create a folder
   2. In terminal
	   1. find your folder
	   2. `git clone https://github.com/citi-onboarding/lgbtqia.git`

1. Open the terminal and go to the folder of this project
2. In terminal:
   1. `git fetch`
   2. `git checkout feature/personalidades`
   3. open the server folder and enter `yarn install`
3. Create a .env file, if you don't already have one, and put in the information available on Squad Dalia notion
4. In terminal:
   1. `docker compose up`
	wait until you see:
	![enter image description here](https://imgur.com/l288stL.png)

5. Test the post, get, delete and put methods (pay special attention to the nullable columns)
6. Test the things that you think are worth testing, even the ones that are not in this description

PS: if you have trouble with docker, clean-restart it with:

    docker compose down
    docker rm -f $(docker ps -a -q)
    docker volume rm $(docker volume ls -q)
    docker-compose up
